R/preprocess.R

```diff
@@ -266,13 +319,24 @@ enw_preprocess_data <- function(obs, by = c(), max_delay = 20,
 
   # assign groups
   obs <- enw_assign_group(obs, by = by)
-
-  # filter by maximum report date
-  obs <- obs[, .SD[report_date <= (reference_date + max_delay - 1)],
+  by_with_group_id <- c("group",by)
+  
+  # complete missing report dates
+  # Note: it might be better to get the "now" date as last date instead of
+  # estimating it from data using the max report date
+  obs <- enw_complete_dates(obs,
+                            min_date = min(obs$reference_date,na.rm=T),
+                            max_date =  max(obs$report_date),
+                            groups = unique(obs[,..by_with_group_id]),
+                            max_delay = max_delay,
+                            include_missing = any(is.na(obs$reference_date)))
```

Comment:
See comment. Should discuss if we want to have something like this. If the dataset does not go until the last report date, we could have a different time window than intended by the user.

Reply:
Yup agree this is an issue. See [#81](https://github.com/epiforecasts/epinowcast/issues/81) for some concerns from @sbfnk about making this be to hardcoded to work on days. 

It is hard to know what the optimal course to take here is. Shall we have this as a step for now and look at changing in the future (though this will break old code) or have this as a function users can use themselves before beginning preprocessing?

Reply:
We had a chat and agreed this is good to keep for now as likely to impact many users. If/when extending to arbitrary timespans this decision will need to be rethought.